cpp/purify/mpi_utilities.h

```diff
@@ -3,10 +3,12 @@
 
 #include "purify/config.h"
 #include <vector>
-#include <sopt/mpi/communicator.h>
 #include <sopt/linear_transform.h>
 #include "purify/utilities.h"
 
+#ifdef PURIFY_MPI
+#include <sopt/mpi/communicator.h>
+#endif
```

Comment:
Why would `mpi_utilites` need guards if is `mpi_utilities`...??? Shouldn't be just used when used with mpi??